packages/keyring-controller/src/KeyringController.ts

```diff
@@ -441,6 +441,20 @@ export class KeyringController extends BaseControllerV2<
     return this.#keyring.getKeyringForAccount(account);
   }
 
+  /**
+   * Returns the encryption public key of
+   * the specified `address` if one exists.
+   *
+   * @deprecated Use of this method is discouraged as actions executed directly on
+   * keyrings are not being reflected in the KeyringController state and not
+   * persisted in the vault.
+   * @param account - An account address.
+   * @returns Promise resolving to encyption public key of the `account` if one exists.
```

Comment:
Nit: we can specify that this method throws an exception when there's no keyring managing that address and when the keyring does not support public key encryption:

```suggestion
   * @returns Promise resolving to encyption public key of the `account` if one exists.
   * @throws If the `account` does not exist or does not support the `getEncryptionPublicKey` method
```